presto-spi/src/main/java/com/facebook/presto/spi/type/NestedField.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Map;
+
+public class NestedField
```

Comment:
If I have a struct like `row_field = row(F1 integer, F2 varchar, F3 varchar, F4 varchar)` and I do a `select row_field.F1 from nested_field_test` I see that this class contains `name=row_field` and `fields` contain only the projected `F1` field. So this class really models a struct with its projected fields, maybe we should rename this one. 

Also this class supports infinitely nested fields, does the current code support it? For ex, if I have a struct 
`(a: int (b: int, c: (d: int)))` does the current code do pruning (and prunes b) if I do `select a.c.d from T`?